src/main/java/com/example/project/model/Book.java

```diff
@@ -0,0 +1,38 @@
+package com.example.project.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+import org.antlr.v4.runtime.misc.NotNull;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Data
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Table(name = "books")
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NotNull
```

Comment:
The @NotNull annotation from org.antlr.v4.runtime.misc.NotNull is not suitable for use in JPA entities. This annotation is part of ANTLR, which is a tool for language recognition, and its @NotNull is not related to JPA or bean validation.

For JPA entities, you should use @NotNull from the javax.validation.constraints package, which is a standard Java Bean Validation (JSR 380) annotation used to indicate that a particular field should not be null. It can be used in combination with @Column(nullable = false) which is a JPA annotation that indicates to the database schema generation process that a particular column should not allow nulls. Fix everywhere, be careful.

Reply:
not fixed